src/scheduler/resource_resv.c

```diff
@@ -1330,13 +1329,10 @@ update_resresv_on_end(resource_resv *resresv, char *job_state)
 			resresv->execselect = NULL;
 		}
 		/* We need to correct our calendar */
-		if (resresv->server->calendar != NULL) {
-			te = find_timed_event(resresv->server->calendar->events, 0, resresv->name, TIMED_END_EVENT, 0);
-			if (te != NULL)
-				set_timed_event_disabled(te, 1);
-		}
+			if (resresv->end_event != NULL)
+				set_timed_event_disabled(resresv->end_event, 1);
```

Comment:
when should one call delete_event() vs set_timed_event_disabled() ?

Reply:
I'm not exactly sure.  The disabled event code is used in degraded reservations.  You disable events to check if you can reconfirm the reservation without losing what you already have.  I don't know tha reconfirming code all that well.  I just kept the code here the same.  Instead of searching for an end event to disable it, I just directly do it.